src/net/sourceforge/kolmafia/session/InventoryManager.java

```diff
@@ -1155,33 +1156,34 @@ public static int itemValue(final AdventureResult item, final boolean exact) {
     return lower + (int) ((upper - lower) * factor);
   }
 
-  public static final int priceToAcquire(
-      final AdventureResult item, int quantity, final boolean exact) {
-    return InventoryManager.priceToAcquire(item, quantity, exact, false, 0);
+  public static final long priceToAcquire(final AdventureResult item, final boolean exact) {
+    return InventoryManager.priceToAcquire(item, exact, false, 0);
   }
 
-  public static final int priceToAcquire(
-      final AdventureResult item, int quantity, final boolean exact, final boolean mallPriceOnly) {
-    return InventoryManager.priceToAcquire(item, quantity, exact, mallPriceOnly, 0);
+  public static final long priceToAcquire(
+      final AdventureResult item, final boolean exact, final boolean mallPriceOnly) {
+    return InventoryManager.priceToAcquire(item, exact, mallPriceOnly, 0);
   }
 
-  private static final int priceToAcquire(
+  private static final long priceToAcquire(
       final AdventureResult item,
-      int quantity,
       final boolean exact,
       final boolean mallPriceOnly,
       final int level) {
-    int price = 0;
-    int needed = quantity;
+
+    int itemId = item.getItemId();
+    int needed = item.getCount();
+
     // Not just inventory; include anything our setting allow to be retrieved
     int onhand = Math.min(needed, InventoryManager.getAccessibleCount(item));
+    long price = 0;
 
     if (onhand > 0) {
-      if (item.getItemId() != ItemPool.PLASTIC_SWORD) {
+      if (itemId != ItemPool.PLASTIC_SWORD) {
```

Comment:
The change here is fine, but why is the TPS special-cased here anyway?

Might be worth adding the comment from the commit history:

    Added a special case to the create vs. buy decision-making code: an on-hand
    tiny plastic sword is valued at 0 meat, since you get it back from the drink.
    This should avoid undesirable behavior if the historical price of the TPS is
    ever greater than one of the items containing a TPS.

Reply:
Yeah. Fortunately, I have the verbose "svn log" output saved, so it was easy to discover this was r8873 by jasonharper from 2011-01-04. I have another comment from him in InventoryManager.itemValue - which I added when I was finding it difficult to understand what that method was calculating. He was very good at writing detailed commit messages. Not so much with in-line comments.

Comment added.